i_home_statistik_cardview

ImSeDanny opened the pull request:
A_Tab_Home:
- Liste mit den 3 aktuellsten Bons implementiert --> wird über Methode "prepareHomeData" aktuallisiert und hinzugefügt
- 3 Dummy-Diagramme implementiert (Kreisdiagramm+ 2x Balkendiagramm)

A_Tab_Bons:
- Liste mit 20 Bons --> Methode "prepareBonData" angepasst: wird davor gecleared und dann befüllt

box_tab1_home_content.xml:
- Cardviews wurden statisch hinzugefügt (nicht dynamisch)
- NestedScrollView wurde hinzugefügt --> dadurch wird die Actionbar beim scrollen gehided

SshaBrn committed:
Cardview testweise erstellt

- PieChart erstellt
- BarChart erstellt

erster Test. Layout muss überarbeitet werden.

SshaBrn committed:
Cardviews mit Recyclerview (i_home_liste_implementieren) zusammengelegt.
Erster Entwurf Layout noch nicht final.

- 3 Cardviews mit Statistiken
- 1 Recyclerview mit den letzten 3 Bons

SshaBrn committed:
Cardviews mit Recyclerview (i_home_liste_implementieren) zusammengelegt.
Erster Entwurf Layout noch nicht final.

- 3 Cardviews mit Statistiken
- 1 Recyclerview mit den letzten 3 Bons

SshaBrn committed:
Cardviews mit Recyclerview (i_home_liste_implementieren) zusammengelegt.
Erster Entwurf Layout noch nicht final.

- 3 Cardviews mit Statistiken
- 1 Recyclerview mit den letzten 3 Bons

dontcallme committed:
tab home statistik card view erstellt, liste mit 3 Charts + Bons darunter

ImSeDanny committed:
Liste wird vor dem Füllen gecleared

A_Tab1Home:
- Methode prepareHomeData:
- Methodennamen renamed: zu prepareHomeData
- for-schleife zähler von 20 auf 3
- Bevor for-schleife ausgeführt wird, wird Liste gecleared

A_Tab3Bons:
- Methode prepareBonData Zähler von 3 auf 20
- Bevor for-schleife ausgeführt wird, wird Liste gecleared

SshaBrn committed:
Nested Scrollview implementiert

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,158 @@
+
+        data.add(new PieEntry(0.2f, 0));
+        data.add(new PieEntry(0.2f, 1));
+        data.add(new PieEntry(0.50f, 2));
```
Ist es nicht möglich für die Befüllung der Diagramme eigene Methoden zu schreiben? Wäre deutlich übersichtlicher und später einfacher im Code zu finden.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,158 @@
+        bChart.setData(data);
+
+
+        //Balkendiagramm
```
Allgemein ist das alles hier ziemlich redundant. Bitte baut Methoden um diese Redundanzen zu verhindern.
Bitte schreibt eine Methode und packt alles von zeile 109 - 123 rein und passt es entsprechend an, z.B.
`private void createChart(BarChart bar, ArrayList<IBarDataSet> daten)`

danach bentzt bitte diese Methode um das zweite chart zu bauen.

schmitzdominic reviewed (changes requested):
Hi,
im großen und ganzen recht gut, nur bitte beachtet das ihr euch viel Arbeit umsonst macht. 
- Bitte baut Methoden wenn euch Code irgendwie redundant vorkommt
- Bitte baut Methoden um Darstellungen mit Daten zu füllen der Übersichtlichkeit halber.

Wenn ihr das annehmbar geändert habt, gibts von mir ein approve

ImSeDanny committed:
Erstellung der Diagramme in Methoden ausgelagert

A_Tab1Home:
- Methode createBarChart erstellt
- erstellt ein Balkendiagramm

- Methode prepareBarData erstellt
- befüllt ein Balkendiagramm mit Werten

- Methode createPieChart erstellt
- erstellt ein Kreisdiagramm

- Methode preparePieDataa erstellt
- befüllt ein Kreisdiagramm mit Werten

ImSeDanny commented:
@schmitzdominic ich hab das ganze gefixed und die Erstellung der verschiedenen Diagramme in Methoden ausgelagert:

A_Tab_Home:
- Methode zur Erstellung eines PieChart
- Methode zur Befüllung eines PieChart
- Methode zur Erstellung eines BarChart
- Methode zur Befüllung eines BarChart

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,146 @@
+
+    /*
+     * Füllt die Values eines Balkendiagramms mit random Werten
+     */
```
Hier auch. Bitte auch beachten, Methoden immer so zu dokumentieren das deren Funktion klar wird. Also bitte einfach das wort random rausnehmen.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,146 @@
+    /*
+     * Erstellt ein Balkendiagramm
+     * Zuerst werden die Werte gefüllt und dann in das Diagramm eingelesen
+     */
```
Bitte /** Kommentar */ verwenden!

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,146 @@
+    /*
+     * Erstellt ein Kreisdiagramm
+     * Zuerst werden die Werte gefüllt und dann in das Diagramm eingelesen
+     */
```
Hier auch

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,146 @@
+
+        //Balkendiagramm 1
+        bChart = (BarChart) rootView.findViewById(R.id.chart2);
+        ArrayList<IBarDataSet> dataSets1 = new ArrayList<IBarDataSet>();
```
Bitte hinten Diamont verwenden! <>

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,146 @@
+
+        BarDataSet set;
+
+        set = new BarDataSet(val, name);
```
Wieso in 2 getrennte Zeilen? Wieso nicht mit zeile 99 in einer Zeile??

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,146 @@
+
+        daten.add(set);
+
+        BarData data = new BarData(daten);
```
geht hier nicht auch `BarData data = new BarData(daten.add(set));`
Dadurch spart ihr euch die Zeile 104

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,146 @@
+        ArrayList<PieEntry> val = new ArrayList<>();
+        preparePieData(val);    // Daten werden gefüllt
+
+        final PieDataSet dataSet = new PieDataSet(val, "Test");
```
Bitte nicht so! Keine Statischen Strings in Methoden definieren. Den Namen kannst du zur not auch als Übergabeparameter definieren

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Tab1Home.java:
```diff
@@ -7,16 +7,146 @@
+     * Füllt die Values eines Balkendiagramms mit random Werten
+     */
+    private void prepareBarData(ArrayList<BarEntry> values){
+        for (int i = (int) 0; i < 10 + 1; i++) {
```
Bitte den explizieten Cast (int) entfernen danke!

schmitzdominic reviewed (changes requested):
Gut das ihr es umgebaut hat, allerdings fehlen noch ein paar sachen :)

ImSeDanny committed:
Fehler gefixed

A_Tab1Home:
- Kommentare richtig erstellt
- Code gekürzt
- expliziten Cast entfernt
- statischen String durch Parameter ersetzt (Name des Charts)

schmitzdominic reviewed (approved):
Gut gemacht :) gibt nen approve ;)

Sippl reviewed (approved):
Läuft bei mir soweit :) Von mir gibts ein Ok
